lib/ansible/modules/cloud/vmware/vmware_guest_find.py

```diff
@@ -99,29 +99,17 @@ def __init__(self, module):
         self.use_instance_uuid = self.params['use_instance_uuid']
 
     def getvm_folder_paths(self):
-        results = []
-        vms = []
-
-        if self.uuid:
-            if self.use_instance_uuid:
-                vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="instance_uuid")
-            else:
-                vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="uuid")
-            if vm_obj is None:
-                self.module.fail_json(msg="Failed to find the virtual machine with UUID : %s" % self.uuid)
-            vms = [vm_obj]
-
+        if self.uuid and self.use_instance_uuid:
+            vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="instance_uuid")
+        elif self.uuid:
+            vm_obj = find_vm_by_id(self.content, vm_id=self.uuid, vm_id_type="uuid")
         elif self.name:
-            objects = self.get_managed_objects_properties(vim_type=vim.VirtualMachine, properties=['name'])
-            for temp_vm_object in objects:
-                if temp_vm_object.obj.name == self.name:
-                    vms.append(temp_vm_object.obj)
+            vm_obj = find_vm_by_name(self.content, vm_name=self.name)
 
-        for vm in vms:
-            folder_path = self.get_vm_path(self.content, vm)
-            results.append(folder_path)
+        if vm_obj is None:
+            self.module.fail_json(msg="Failed to find the virtual machine with UUID: %s, name: %s" % (self.uuid, self.name))
```

Comment:
`self.uuid` is None when name is specified.

Reply:
I did that on purpose. I would prefer to only have one error message.